app/policies/trainee_policy.rb

```diff
@@ -60,7 +60,7 @@ def update?
   end
 
   def withdraw?
-    write? && (defer? || trainee.deferred?)
+    write? && (defer? || trainee.deferred? || trainee.withdrawn?)
```

Comment:
I think `user_is_system_admin?` would be preferred instead of `trainee.withdrawn?`. It makes more sense to me that a trainee can be allowed withdraw actions even though the trainee is already withdrawn but is an admin if that makes sense.